### DISCUSSION TITLE
adding github actions / workflows for a homebrew tap

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#1846] [Tap maintenance and Homebrew development]
i spent an hour or two attempting to build some bottles for a homebrew tap i have forked. i experimented with several different yml file(s) / workflows for my tap in hopes of building some bottles of the formula that are contained within the tap.  i'm using macos mojave as a `self-hosted` runner, and kept running into authentication issues when running a workflow in either service mode or starting the self-hosted runner in the foreground of an active terminal session.

it would prompt for my username and password.

i tried my username, ie. `ipatch` along with my password, my PAT, and the token I used to setup the self hosted runner, and all three entries would fail to authenticate. the failure seems to begin when my workflow attempts to setup homebrew.

my fork of the homebrew-freecad repo can be found below

https://github.com/ipatch/homebrew-freecad

i have a created the branch

https://github.com/ipatch/homebrew-freecad/tree/ipatch.exp.with.github.actions

where i've been editing the workflow files without much success. i'm obviously missing something here, and assistance with this issue would greatly be appreciated. 🙏

**Comment by carlocab** [comment]
Can you get it to work with a non-self-hosted GitHub runner? It sounds to me like you should try that first to make sure your problems aren't somewhere there, and then look into getting your self-hosted runner working. It might also be simpler to test your self-hosted runner with a smaller workflow just to see it does what you're expecting it to.

**Reply by ipatch** [reply]
it appears it's something related with the homebrew/actions, as i just changed the runner param,

from

```
runs-on: self-hosted
```

to

```
runs-on: macos-10.14
```

and the workflow is bumping along. 🤷‍♂️

https://github.com/ipatch/homebrew-freecad/runs/3100980402

ie.

the runner task did not prompt for a username and password, and there were no authentication issues.

**Reply by carlocab** [reply]
Are you able to run a non-Homebrew-test-bot workflow on your self-hosted runner?

**Reply by ipatch** [reply]
that i have not attempted.

do you have a particular workflow i should attempt to run?

**Reply by carlocab** [reply]
You should probably attempt that first to determine if your self-hosted runner is working correctly.

I don't have any suggestions as to example workflows to run. You can probably check the GitHub docs/tutorials for basic examples.
